app/src/main/java/com/powerhungers/fomezero/feature/registration/presentation/RegistrationFragment.kt

```diff
@@ -25,16 +29,30 @@ class RegistrationFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        handleClickListener()
+        checkRadioButton(binding.radioGroup)
         addObserver()
     }
 
-    private fun handleClickListener() {
+    private fun checkRadioButton(radio: RadioGroup) {
+
+        radio.setOnCheckedChangeListener { _, checkedId ->
+            when (checkedId) {
+                R.id.consumerRadioButton -> {
+                    handleClickListener("consumer")
+                }
+                R.id.producerRadioButton -> {
+                    handleClickListener("producer")
+                }
+            }
+        }
+    }
```

Comment:
```suggestion
enum class UserType(val value : String) {
   PRODUCER("produtor"),
   CONSUMER("consumidor")
}

    private fun checkRadioButton() : UserType {
        binding.radio.setOnCheckedChangeListener { _, checkedId ->
           return when (checkedId) {
                R.id.consumerRadioButton -> 
                   UserType.CONSUMER
                R.id.producerRadioButton -> 
                    UserType.PRODUCER
            }
        }
    }
```
Essa enum class ja existe no projeto e esta no arquivo de response do user, vc deve renome a la para UserTypeResponse e criar um UserType no pacote de domain para poder utiliza la em sua fragment